collector/netdev_common.go

```diff
@@ -114,13 +117,23 @@ func (c *netDevCollector) Update(ch chan<- prometheus.Metric) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get netstats: %w", err)
 	}
+
+	netDevInterfaces, err := getNetDevInterfaces()
```

Comment:
This is going to scan sysfs even if `netdevLabelIfAlias` is false. I think what we actually need to do here is simplify this down to a single function that returns a `map[string][string]string`.

Where the map is `map[interface][label]labelValue`.